airflow/models/dagrun.py

```diff
@@ -626,6 +625,7 @@ def _emit_duration_stats_for_finished_state(self):
 
     @provide_session
     def verify_integrity(self, session: Session = None):
+        from airflow.settings import task_instance_mutation_hook
```

Comment:
Avoid import in function definitions. https://docs.python.org/3.1/howto/doanddont.html#from-module-import-name1-name2
```suggestion
```

Reply:
Let's change L642 to `settings.task_instance_mutation_hook(ti)` in that case

Reply:
> Avoid import in function definitions. https://docs.python.org/3.1/howto/doanddont.html#from-module-import-name1-name2

That is talking about `import *` inside functions. This is not what we are doing here.